src/praisonai-bot/praisonai_bot/scheduler/executor.py

```
@@ -0,0 +1,598 @@
+        # When it reports "nothing to do" the tick is recorded as ``skipped`` —
+        # no tokens spent, no empty message delivered.  When it reports "go"
+        # any context it produced is appended to the message before the turn.
+        gate = self._resolve_condition(job)
+        if gate is not None:
+            try:
+                # Run off the event loop: the default gate shells out via
+                # subprocess.run(), which would otherwise block every other
+                # tick / delivery / health-check coroutine for the gate's
+                # timeout. Mirrors the agent.chat() offload below.
+                decision = await asyncio.to_thread(gate.should_run, job)
+            except Exception as e:  # pragma: no cover - defensive
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)